x/ccv/provider/keeper/relay.go

```diff
@@ -73,21 +73,12 @@ func (k Keeper) HandleLeadingVSCMaturedPackets(ctx sdk.Context) {
 // TODO: Unit test this method.
 func (k Keeper) HandleVSCMaturedPacket(ctx sdk.Context, chainID string, data ccv.VSCMaturedPacketData) {
 	// iterate over the unbonding operations mapped to (chainID, data.ValsetUpdateId)
-	unbondingOps, _ := k.GetUnbondingOpsFromIndex(ctx, chainID, data.ValsetUpdateId)
 	var maturedIds []uint64
-	for _, unbondingOp := range unbondingOps {
-		// remove consumer chain ID from unbonding op record
-		unbondingOp.UnbondingConsumerChains, _ = removeStringFromSlice(unbondingOp.UnbondingConsumerChains, chainID)
-		k.Logger(ctx).Debug("unbonding operation matured on consumer", "chainID", chainID, "opID", unbondingOp.Id)
-
-		// If unbonding op is completely unbonded from all relevant consumer chains
-		if len(unbondingOp.UnbondingConsumerChains) == 0 {
+	for _, unbondingOp := range k.GetUnbondingOpsFromIndex(ctx, chainID, data.ValsetUpdateId) {
+		// Remove consumer chain ID from unbonding op record
+		if k.RemoveConsumerFromUnbondingOp(ctx, unbondingOp.Id, chainID) {
```

Comment:
I'm admittedly not very familiar with unbonding op logic, but imo it's notable that `HandleVSCMaturedPacket` is not unit tested. Is it worth adding the missing unit test to validate we're not regressing production code with the refactor?

Reply:
Alternative: we can save this refactor for a separate PR which also adds the UT. I'm just hesitant to refactor important logic like this that's not unit tested, esp because I don't exactly understand the desired functionality